lib/dns.js

```diff
@@ -65,11 +65,9 @@ function onlookup(err, addresses) {
   if (err) {
     return this.callback(dnsException(err, 'getaddrinfo', this.hostname));
   }
-  if (this.family) {
-    this.callback(null, addresses[0], this.family);
-  } else {
-    this.callback(null, addresses[0], isIP(addresses[0]));
-  }
+  this.callback(null, addresses[0], this.family ?
+    this.family :
+    isIP(addresses[0]));
```

Comment:
try this?
```js
  this.callback(null, addresses[0], this.famliy ?? isIP(addresses[0]));
```

Reply:
I checked and found that the internal code does not use the `Nullish Coalescing` syntax, I am not quite sure whether the specification allows.

Reply:
or `||`, just a suggestion

Reply:
we do use `||`

https://github.com/nodejs/node/blob/18c3ff99aa26064199c75e220145a465980cf82b/lib/dns.js#L86

Reply:
👍 Yes, this is more concise.

Reply:
@Himself65 We *could* use nulllish coalescing in the code base, but it:

- Breaks the doc generator for now and that would have to be resolved first, see https://github.com/nodejs/node/pull/32628
- Cannot be backported to older release lines

But most importantly, it is **not** equivalent to `||` here because `family` can be `0`. `||` is correct, `??` isn’t.

Reply:
👍Thank you for explaining.